app/Auth/Authorization.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Auth;
+
+use App\Legacy\Legacy;
+use App\Models\Logs;
+use App\Models\User;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Session;
+use RuntimeException;
+
+class Authorization extends Auth
+{
+	public static function id(): int|null
+	{
+		return is_int(parent::id()) ? parent::id() : null;
+	}
+
+	public static function isAdmin(): bool
+	{
+		return parent::user()?->isAdmin() === true;
+	}
+
+	/**
+	 * @return bool
+	 */
+	public static function canUpload(): bool
+	{
+		$user = parent::user();
+
+		return $user?->id === 0 || $user?->may_upload === true;
+	}
+
+	/**
+	 * Returns true if the user matches the id or if is admin.
+	 *
+	 * @param int $id to check
+	 *
+	 * @return bool
+	 */
+	public static function isCurrentOrAdmin(int $id)
+	{
+		return self::id() === 0 || self::id() === $id;
+	}
+
+	/**
+	 * Sets the session values when no there is no username and password in the database.
+	 *
+	 * @return bool returns true when no login was found
+	 */
+	public static function noLogin(): bool
+	{
+		/** @var User|null $adminUser */
+		$adminUser = User::query()->find(0);
+		if ($adminUser !== null && $adminUser->password === '' && $adminUser->username === '') {
+			parent::login($adminUser);
+
+			return true;
+		}
+
+		return Legacy::noLogin();
+	}
+
+	/**
+	 * @return void
+	 *
+	 * @throws RuntimeException
+	 */
+	public static function logout()
+	{
+		parent::logout();
+		Session::flush();
```

Comment:
Laravel recommends invalidating the session and regenerating the CSRF tokens.
https://laravel.com/docs/9.x/authentication#logging-out

```
public function logout(Request $request)
{
    Auth::logout();
 
    $request->session()->invalidate();
 
    $request->session()->regenerateToken();
 
    return redirect('/');
}
```

Reply:
Understood. But this is not what we are doing, are we? We do not invalidate the session and regenerate the token but throw away the whole session.

Maybe, we should just keep it that way, because we have always done it this way (and so this PR does not change too much), but I still feel like we should do it as expected by the Laravel documentation (at least in the long run) and I wonder if there is any technical reason why we do it differently and if there is anything which would break if we do it as recommended by the docs.